Automattic/wp-calypso
# User mentions: connect to real suggestions

Adds the user suggestions higher-order component to our comment form, which provides real suggestions from the API based on the current post's site.

<img width="339" alt="screen shot 2018-05-03 at 12 05 06 pm" src="https://user-images.githubusercontent.com/17325/39555038-4ef619da-4eca-11e8-962f-84a686894ed2.png">

Fixes #24525.
Fixes https://github.com/Automattic/wp-calypso/issues/24523 too.

### To test

Head to a Reader full post on a site with lots of users (like an internal P2). Hit the @ symbol and try choosing a user.

For known issues, check https://github.com/Automattic/wp-calypso/milestone/239.
